polyfill/lib/regex.mjs

```diff
@@ -1,7 +1,21 @@
+const tzComponentNotBCP47 = new RegExp(
+  [
+    '\\.\\.[-A-Za-z._]{1,12}',
+    '\\.[-A-Za-z_][-A-Za-z._]{0,12}',
+    '_[-A-Za-z._]{0,13}',
+    '[a-zA-Z](?:[A-Za-z._][-A-Za-z._]{0,12})?',
+    '[a-zA-Z]-(?:[-._][-A-Za-z._]{0,11})?',
+    '[a-zA-Z]-[a-zA-Z](?:[-._][-A-Za-z._]{0,10})?',
+    '[a-zA-Z]-[a-zA-Z][a-zA-Z](?:[A-Za-z._][-A-Za-z._]{0,9})?',
+    '[a-zA-Z]-[a-zA-Z][a-zA-Z]-(?:[-._][-A-Za-z._]{0,8})?',
+    '[a-zA-Z]-[a-zA-Z][a-zA-Z]-[a-zA-Z](?:[-._][-A-Za-z._]{0,7})?',
+    '[a-zA-Z]-[a-zA-Z][a-zA-Z]-[a-zA-Z][a-zA-Z](?:[-._][-A-Za-z._]{0,6})?'
+  ].join('|')
+);
```

Comment:
Which grammar describes this? For such a large regex, a spec link in the source would be nice.

Reply:
This is described in https://github.com/tc39/proposal-temporal/pull/1325/files#diff-3a221d4b1eb72b9134f63736e30ed2d2c11eccd7ee2fb34beb3c9ac8e1c3247cR973 - the "but not" part. I'll add a comment.